File: scripts/algorithms/M/Maximum Twin Sum of a Linked List/Maximum Twin Sum of a Linked List.js
var pairSum = function(head) {
    const arr = [];
    let max = 0;
    
    while (head) {
        arr.push(head.val);
        head = head.next;
    }
    
    for (let i = 0; i < arr.length / 2; i++) {
        const sum = arr[i] + arr[arr.length - 1 - i]
        max = Math.max(max, sum);
    }
    
    return max;
};

File: scripts/algorithms/M/Maximum Twin Sum of a Linked List/Maximum Twin Sum of a Linked List.py
# Runtime: 2246 ms (Top 8.54%) | Memory: 54.1 MB (Top 58.68%)
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        nums = []
        curr = head
        while curr:
            nums.append(curr.val)
            curr = curr.next

        N = len(nums)
        res = 0
        for i in range(N // 2):
            res = max(res, nums[i] + nums[N - i - 1])
        return res